packages/pds/src/api/com/atproto/identity/updateHandle.ts

```diff
@@ -20,28 +20,39 @@ export default function (server: Server, ctx: AppContext) {
       })
 
       // Pessimistic check to handle spam: also enforced by updateHandle() and the db.
-      const available = await ctx.services
-        .account(ctx.db)
-        .isHandleAvailable(handle)
-      if (!available) {
+      const handleDid = await ctx.services.account(ctx.db).getHandleDid(handle)
+      if (handleDid !== requester) {
         throw new InvalidRequestError(`Handle already taken: ${handle}`)
       }
 
-      const seqHandleTok = await ctx.db.transaction(async (dbTxn) => {
-        let tok: HandleSequenceToken
-        try {
-          tok = await ctx.services
-            .account(dbTxn)
-            .updateHandle(requester, handle)
-        } catch (err) {
-          if (err instanceof UserAlreadyExistsError) {
-            throw new InvalidRequestError(`Handle already taken: ${handle}`)
-          }
-          throw err
+      let seqHandleTok: HandleSequenceToken
+      if (handleDid) {
+        if (handleDid !== requester) {
+          throw new InvalidRequestError(`Handle already taken: ${handle}`)
```

Comment:
This is the same check as above, right?

Reply:
oh oops - i meant to only have one of these